app/uk/gov/hmrc/apipublisher/models/PublisherRequest.scala

```diff
@@ -20,13 +20,23 @@ import play.api.libs.json._
 import uk.gov.hmrc.apipublisher.models.APICategory.{APICategory, formatAPICategory}
 import uk.gov.hmrc.http.UnprocessableEntityException
 
+import scala.collection.immutable
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
 case class ApiAndScopes(api: JsObject, scopes: JsArray) {
 
-  def validateAPIScopesAreDefined(): Unit = {
-    val missing: Seq[String] = apiScopes.filterNot(definedScopes.contains)
-    if (missing.nonEmpty) {
-      throw new UnprocessableEntityException(s"Undefined scopes used in definition: ${missing.mkString("[", ", ", "]")}")
-    }
+  def validateAPIScopesAreDefined(storedScopes: Future[Seq[Scope]] = Future(Seq())): Future[ScopesDefinedResult] = {
```

Comment:
Don't pass Futures as parameters to functions pls

Reply:
This function appears to only work on the data that is passed to it.
If that is the case it would better sit in a companion object.